common/test/java/com/couchbase/lite/internal/core/C4CollectionDocObserverTest.kt

```diff
@@ -0,0 +1,104 @@
+package com.couchbase.lite.internal.core
+
+import com.couchbase.lite.AbstractIndex
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertNotNull
+import org.junit.Ignore
+import org.junit.Test
+import java.util.concurrent.atomic.AtomicInteger
+
+class C4CollectionDocObserverTest : C4BaseTest() {
+    private val mockCollectionDocObserver = object : C4CollectionDocObserver.NativeImpl {
+        override fun nCreate(coll: Long, docId: String?): Long = 0xd8597341L
+        override fun nFree(peer: Long) = Unit
+    }
+
+    @Test
+    fun testCreateC4CollectionDocObserver() {
+        val impl = C4CollectionDocObserver.nativeImpl
```

Comment:
I don't believe there is any need for this.  This is what the `create` that takes a native impl as parameter, is for.